common/core/nesteddisplay.cpp

```diff
@@ -22,16 +22,145 @@
 #include <sstream>
 
 #include <hwctrace.h>
+#include <hwclayer.h>
+#include "utils_android.h"
 
 namespace hwcomposer {
+#ifdef NESTED_DISPLAY_SUPPORT
```

Comment:
Can you squash this and virtual display patch together ?

Reply:
I suggest split these two implement. If we choose using virtual display implement, then we don't need nested display implement. We can merge the nested display based implement first.

Reply:
Can we make that decision first ? Probably send in a PR ontop of this and we can look at how easy/complex it is and make a decision ?

Reply:
buffer sharing is urgent feature in schedule, previous nested display implement has passed SI's fully test, it is proved stable. 
But to adjust the new added nested display, we need modify AOSP code to support it. To avoid change the android AOSP, we propose  a new method using Virtual display, I finished a rough prototype, and it also works, but to improve it, need more efforts.
My plan is that, to catch the schedule, we can use nested display implement first, meanwhile I will improve the Virtual display implement and send to you review. And you will decide which implement will be choose at last.

Reply:
k added final set of comments, if u can address them and we should be able to merge this.